2024-08-20 20:34:39.942 +08:00 [INF] Starting up the application
2024-08-20 20:34:41.455 +08:00 [INF] User profile is available. Using 'C:\Users\pritp\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2024-08-20 20:34:42.030 +08:00 [INF] Now listening on: https://localhost:59507
2024-08-20 20:34:42.031 +08:00 [INF] Now listening on: http://localhost:59508
2024-08-20 20:34:42.036 +08:00 [INF] Application started. Press Ctrl+C to shut down.
2024-08-20 20:34:42.037 +08:00 [INF] Hosting environment: Development
2024-08-20 20:34:42.038 +08:00 [INF] Content root path: C:\Users\pritp\OneDrive\Desktop\bkpFiles\GitBackEnd\FrontEndWebApi\WebApi
2024-08-20 20:34:43.396 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/ - -
2024-08-20 20:34:43.479 +08:00 [INF] HTTP GET / responded 404 in 28.7781 ms
2024-08-20 20:34:43.511 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/ - - - 404 0 - 118.2975ms
2024-08-20 20:34:47.891 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:34:47.901 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:34:47.930 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:35:01.141 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:35:17.588 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:35:30.183 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'WebApiApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:35:30.296 +08:00 [ERR] Failed executing DbCommand (139ms) [Parameters=[@p0='?' (Size = 4000), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_product] ([Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3);
SELECT [ExceptionId]
FROM [db_product]
WHERE @@ROWCOUNT = 1 AND [ExceptionId] = scope_identity();
2024-08-20 20:35:30.355 +08:00 [ERR] An exception occurred in the database while saving changes for context type 'WebApi.Exceptions.Data.LoggingDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2024-08-20 20:38:58.369 +08:00 [INF] An error occurred while saving the entity changes. See the inner exception for details.
2024-08-20 20:38:58.527 +08:00 [INF] Executed DbCommand (10ms) [Parameters=[], CommandType='"Text"', CommandTimeout='30']
SELECT [d].[product_id], [d].[created_at], [d].[created_by], [d].[product_name], [d].[product_description], [d].[updated_at], [d].[updated_by]
FROM [db_product] AS [d]
2024-08-20 20:39:00.518 +08:00 [INF] Executing OkObjectResult, writing value of type 'System.Collections.Generic.List`1[[WebApi.Domain.Products, WebApi.Domain, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'.
2024-08-20 20:39:00.571 +08:00 [INF] Executed action WebApi.Controllers.ProductsController.GetProducts (WebApi) in 252628.5168ms
2024-08-20 20:39:00.578 +08:00 [INF] Executed endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:39:00.584 +08:00 [INF] HTTP GET /api/products/GetProducts responded 200 in 252689.2766 ms
2024-08-20 20:39:00.603 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/api/products/GetProducts - - - 200 - application/json;+charset=utf-8 252711.6851ms
2024-08-20 20:39:00.618 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/favicon.ico - -
2024-08-20 20:39:00.622 +08:00 [INF] HTTP GET /favicon.ico responded 404 in 0.7661 ms
2024-08-20 20:39:00.626 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/favicon.ico - - - 404 0 - 7.9148ms
2024-08-20 20:39:03.722 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:39:03.727 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:39:03.728 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:39:06.761 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:40:55.135 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:40:59.954 +08:00 [ERR] Failed executing DbCommand (3,756ms) [Parameters=[@p0='?' (Size = 4000), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_product] ([Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3);
SELECT [ExceptionId]
FROM [db_product]
WHERE @@ROWCOUNT = 1 AND [ExceptionId] = scope_identity();
2024-08-20 20:40:59.955 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'WebApiApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:40:59.963 +08:00 [ERR] An exception occurred in the database while saving changes for context type 'WebApi.Exceptions.Data.LoggingDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2024-08-20 20:40:59.965 +08:00 [INF] Executed DbCommand (1ms) [Parameters=[], CommandType='"Text"', CommandTimeout='30']
SELECT [d].[product_id], [d].[created_at], [d].[created_by], [d].[product_name], [d].[product_description], [d].[updated_at], [d].[updated_by]
FROM [db_product] AS [d]
2024-08-20 20:41:58.106 +08:00 [INF] An error occurred while saving the entity changes. See the inner exception for details.
2024-08-20 20:41:59.592 +08:00 [INF] Executing OkObjectResult, writing value of type 'System.Collections.Generic.List`1[[WebApi.Domain.Products, WebApi.Domain, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'.
2024-08-20 20:41:59.597 +08:00 [INF] Executed action WebApi.Controllers.ProductsController.GetProducts (WebApi) in 175866.1165ms
2024-08-20 20:41:59.601 +08:00 [INF] Executed endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:41:59.603 +08:00 [INF] HTTP GET /api/products/GetProducts responded 200 in 175877.4231 ms
2024-08-20 20:41:59.606 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/api/products/GetProducts - - - 200 - application/json;+charset=utf-8 175883.5113ms
2024-08-20 20:42:50.279 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:42:50.287 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:42:50.288 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:43:29.511 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:44:50.688 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:44:56.510 +08:00 [ERR] Failed executing DbCommand (4,629ms) [Parameters=[@p0='?' (Size = 4000), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_product] ([Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3);
SELECT [ExceptionId]
FROM [db_product]
WHERE @@ROWCOUNT = 1 AND [ExceptionId] = scope_identity();
2024-08-20 20:44:56.516 +08:00 [ERR] An exception occurred in the database while saving changes for context type 'WebApi.Exceptions.Data.LoggingDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
Invalid column name 'ExceptionId'.
Invalid column name 'ExceptionId'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:718f0a91-a54a-43fe-b3f7-8d04ebce66a0
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2024-08-20 20:46:22.127 +08:00 [INF] Starting up the application
2024-08-20 20:46:23.359 +08:00 [INF] User profile is available. Using 'C:\Users\pritp\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2024-08-20 20:46:23.822 +08:00 [INF] Now listening on: https://localhost:59507
2024-08-20 20:46:23.823 +08:00 [INF] Now listening on: http://localhost:59508
2024-08-20 20:46:23.828 +08:00 [INF] Application started. Press Ctrl+C to shut down.
2024-08-20 20:46:23.830 +08:00 [INF] Hosting environment: Development
2024-08-20 20:46:23.831 +08:00 [INF] Content root path: C:\Users\pritp\OneDrive\Desktop\bkpFiles\GitBackEnd\FrontEndWebApi\WebApi
2024-08-20 20:46:25.303 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/ - -
2024-08-20 20:46:25.374 +08:00 [INF] HTTP GET / responded 404 in 26.2957 ms
2024-08-20 20:46:25.411 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/ - - - 404 0 - 113.3095ms
2024-08-20 20:46:41.758 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:46:41.772 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:46:41.793 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:46:42.068 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:48:35.396 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:48:37.934 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'WebApiApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:48:38.165 +08:00 [ERR] Failed executing DbCommand (200ms) [Parameters=[@p0='?' (DbType = Int32), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (Size = 4000), @p4='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_product] ([ExceptionId], [Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3, @p4);
2024-08-20 20:48:38.221 +08:00 [ERR] An exception occurred in the database while saving changes for context type 'WebApi.Exceptions.Data.LoggingDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'ExceptionId'.
Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:6246cefc-c323-444e-ad31-68e15f666c70
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'ExceptionId'.
Invalid column name 'Message'.
Invalid column name 'Source'.
Invalid column name 'StackTrace'.
Invalid column name 'Timestamp'.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:6246cefc-c323-444e-ad31-68e15f666c70
Error Number:207,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2024-08-20 20:52:47.009 +08:00 [INF] Starting up the application
2024-08-20 20:52:48.138 +08:00 [INF] User profile is available. Using 'C:\Users\pritp\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2024-08-20 20:52:48.551 +08:00 [INF] Now listening on: https://localhost:59507
2024-08-20 20:52:48.552 +08:00 [INF] Now listening on: http://localhost:59508
2024-08-20 20:52:48.557 +08:00 [INF] Application started. Press Ctrl+C to shut down.
2024-08-20 20:52:48.559 +08:00 [INF] Hosting environment: Development
2024-08-20 20:52:48.559 +08:00 [INF] Content root path: C:\Users\pritp\OneDrive\Desktop\bkpFiles\GitBackEnd\FrontEndWebApi\WebApi
2024-08-20 20:52:50.239 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/ - -
2024-08-20 20:52:50.320 +08:00 [INF] HTTP GET / responded 404 in 32.8449 ms
2024-08-20 20:52:50.351 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/ - - - 404 0 - 115.0937ms
2024-08-20 20:53:04.614 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:53:04.622 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:53:04.646 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:53:04.904 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:53:14.010 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:53:15.407 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'WebApiApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:53:15.707 +08:00 [ERR] Failed executing DbCommand (315ms) [Parameters=[@p0='?' (DbType = Int32), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (Size = 4000), @p4='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_ExceptionLog] ([ExceptionId], [Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3, @p4);
2024-08-20 20:53:15.707 +08:00 [INF] Executed DbCommand (8ms) [Parameters=[], CommandType='"Text"', CommandTimeout='30']
SELECT [d].[product_id], [d].[created_at], [d].[created_by], [d].[product_name], [d].[product_description], [d].[updated_at], [d].[updated_by]
FROM [db_product] AS [d]
2024-08-20 20:53:15.768 +08:00 [ERR] An exception occurred in the database while saving changes for context type 'WebApi.Exceptions.Data.LoggingDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert explicit value for identity column in table 'db_ExceptionLog' when IDENTITY_INSERT is set to OFF.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:26d85251-53ea-4177-9462-79d77a85ea76
Error Number:544,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert explicit value for identity column in table 'db_ExceptionLog' when IDENTITY_INSERT is set to OFF.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__188_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:26d85251-53ea-4177-9462-79d77a85ea76
Error Number:544,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2024-08-20 20:53:38.172 +08:00 [INF] Starting up the application
2024-08-20 20:53:39.356 +08:00 [INF] User profile is available. Using 'C:\Users\pritp\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2024-08-20 20:53:39.831 +08:00 [INF] Now listening on: https://localhost:59507
2024-08-20 20:53:39.832 +08:00 [INF] Now listening on: http://localhost:59508
2024-08-20 20:53:39.838 +08:00 [INF] Application started. Press Ctrl+C to shut down.
2024-08-20 20:53:39.839 +08:00 [INF] Hosting environment: Development
2024-08-20 20:53:39.840 +08:00 [INF] Content root path: C:\Users\pritp\OneDrive\Desktop\bkpFiles\GitBackEnd\FrontEndWebApi\WebApi
2024-08-20 20:53:41.343 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/ - -
2024-08-20 20:53:41.403 +08:00 [INF] HTTP GET / responded 404 in 22.1898 ms
2024-08-20 20:53:41.435 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/ - - - 404 0 - 96.9063ms
2024-08-20 20:53:45.974 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/api/products/GetProducts - -
2024-08-20 20:53:45.988 +08:00 [INF] Executing endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:53:46.010 +08:00 [INF] Route matched with {action = "GetProducts", controller = "Products"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetProducts() on controller WebApi.Controllers.ProductsController (WebApi).
2024-08-20 20:53:46.276 +08:00 [INF] Denominator cannot be zero.
2024-08-20 20:53:54.592 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'LoggingDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:53:56.055 +08:00 [INF] Entity Framework Core 6.0.0 initialized 'WebApiApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer:6.0.0' with options: None
2024-08-20 20:53:56.221 +08:00 [INF] Executed DbCommand (176ms) [Parameters=[@p0='?' (Size = 4000), @p1='?' (Size = 4000), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [db_ExceptionLog] ([Message], [Source], [StackTrace], [Timestamp])
VALUES (@p0, @p1, @p2, @p3);
SELECT [ExceptionId]
FROM [db_ExceptionLog]
WHERE @@ROWCOUNT = 1 AND [ExceptionId] = scope_identity();
2024-08-20 20:53:56.325 +08:00 [INF] Executed DbCommand (4ms) [Parameters=[], CommandType='"Text"', CommandTimeout='30']
SELECT [d].[product_id], [d].[created_at], [d].[created_by], [d].[product_name], [d].[product_description], [d].[updated_at], [d].[updated_by]
FROM [db_product] AS [d]
2024-08-20 20:53:56.389 +08:00 [INF] Executing OkObjectResult, writing value of type 'System.Collections.Generic.List`1[[WebApi.Domain.Products, WebApi.Domain, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'.
2024-08-20 20:53:56.436 +08:00 [INF] Executed action WebApi.Controllers.ProductsController.GetProducts (WebApi) in 10414.8631ms
2024-08-20 20:53:56.443 +08:00 [INF] Executed endpoint 'WebApi.Controllers.ProductsController.GetProducts (WebApi)'
2024-08-20 20:53:56.453 +08:00 [INF] HTTP GET /api/products/GetProducts responded 200 in 10475.6644 ms
2024-08-20 20:53:56.477 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/api/products/GetProducts - - - 200 - application/json;+charset=utf-8 10502.5462ms
2024-08-20 20:53:56.491 +08:00 [INF] Request starting HTTP/2 GET https://localhost:59507/favicon.ico - -
2024-08-20 20:53:56.492 +08:00 [INF] HTTP GET /favicon.ico responded 404 in 0.5555 ms
2024-08-20 20:53:56.495 +08:00 [INF] Request finished HTTP/2 GET https://localhost:59507/favicon.ico - - - 404 0 - 3.8643ms
